### DISCUSSION TITLE
Sometimes - Issue with uploading images, S3 storage

### DISCUSSION THREAD

**Discussion by botzill** [directus/directus#17668] [Archive]
Hi.

I have a wired issue that appears from time to time. We store some image to directus and we use S3 as a storage, for 99% of the cases all works OK. But there are 1% of the cases when at some point images are not uploaded but entry in Files is created, here is an example:

<img width="878" alt="Screenshot 2023-03-01 at 17 02 47" src="https://user-images.githubusercontent.com/4686040/222178880-2a7b73bd-47b2-4076-a5ad-9d71c06dbae4.png">

As you can see filename disk is empty for such files. Not sure why we get this issue. We had this in version: [v9.23.0](https://github.com/directus/directus/releases/tag/v9.23.0) and as well  [v9.23.0](https://github.com/directus/directus/releases/tag/v9.23.0)

Our directus is deployed in kubernetes and we running multiple replicas, nginx-ingress.

Once we reload the directus all works ok for some time, 4-5 hours. After that we get the same issue for some of the files. 

Does anyone experience this issue or have any ideas what can be related to?

Thanks a lot guys.

**Comment by ztob** [comment]
hi,

I'm from the same team.
When this issue start to happen that requests to upload image are stuck for a long time then after that will throw with 504 Gateway Timeout

**Comment by rokdd** [comment]
I have the same problem but after reading some amount of images: https://github.com/directus/directus/discussions/17690 . My feeling is that the driver has some problems, because the logs at S3 side did not indicate any errors.

**Reply by elias-summermatter** [reply]
Yes, i would assume the same because we find this error in our logs too.

**Comment by elias-summermatter** [comment]
Same issue here

**Comment by botzill** [comment]
I hope directus team will see this and give us more insight

**Comment by rijkvanzanten** [comment]
@botzill I'm curious if it's caused by the same underlying cause as some perf issues we were seeing on a lot of reads:

https://github.com/aws/aws-sdk-js-v3/issues/3560#issuecomment-1484140333

From what I've gathered so far, it seems like v3 of the S3 SDK fails to clean up sockets that it failed to recognize as closed, which in turn causes the socket queue to fill up, which in turn causes any operation to/from S3 to completely hang
